client/components/Layout/LayoutBanner.tsx

```diff
@@ -74,10 +74,11 @@ class LayoutBanner extends Component<Props, State> {
 			this.props.content.buttonType || (this.props.content.showButton && 'create-pub');
 
 		const buttonText = getButtonText(buttonType, this.props.content.buttonText, isLoggedIn);
+
 		const buttonUrl =
 			buttonType === 'link'
 				? this.props.content.buttonUrl
-				: isLoggedIn && buttonType === 'create-pub'
+				: !isLoggedIn && buttonType === 'create-pub'
```

Comment:
These are just my style opinions, so feel free to ignore, but I liked your earlier commit that was unwinding these conditional ternaries. In practice, I think they're hard to read and it's really easy to lose track, especially with this 

IMO it's actually worth modifying this to be a little more explicit (either with parentheses, a more expressive indentation, or splitting it into a few separate conditional steps like you had done) to help prevent future mixups here since we don't have tests covering this.

Reply:
Also if you do go back to the if/else version, I think you can drop that final `else` clause since after `let buttonUrl`, `buttonUrl` will already have the value `undefined`